### PR TITLE
Make smooth tooltip

### PR DESCRIPTION
**Pull request by smyzin**
- removed `react-tooltip` lib
- styled owned tooltip
- created `click outside` hook

### CODE REVIEW

File: src/components/ZkProofButton.tsx [hunk 1]
```diff
@@ -48,18 +63,31 @@ const QuestionMark = () => {
...
+  useClickOutside(zkProofButtonRef, () => setIsShow(false))
+
   return (
-    <>
-      <div className={zkProofButton} data-tip={popoverText}>
+    <div className={zkProofButtonTooltip}>
+      <div className={tooltipWrapper}>
+        <ToolTip text={popoverText} show={isShow} />
+      </div>
+      <div
```
Comment (MixailE) [inline src/components/ZkProofButton.tsx:79]: Can we put the logic in `ToolTip`?

Comment (smyzin) [reply src/components/ZkProofButton.tsx]: Did it. Thx

File: src/components/ToolTip.tsx [hunk 2]
```diff
@@ -1,45 +1,91 @@
...
+    position('absolute'),
     backgroundColor('bg-formal-accent'),
-    maxWidth('max-w-sm'),
-    mobile ? inset('!left-0', '!right-0') : undefined,
-    opacity('!opacity-100'),
-    borderRadius('!rounded-lg')
+    inset(mobile ? 'left-0' : 'left-3', mobile ? 'right-0' : 'right-3'),
+    inset('bottom-4'),
+    padding('py-2', 'px-5'),
+    margin('!ml-0'),
```
Comment (T-Damer) [inline src/components/ToolTip.tsx:44]: Do we need important here if we don't override original library?

File: src/components/ToolTip.tsx [hunk 3]
```diff
@@ -1,45 +1,91 @@
...
+    zIndex('z-50'),
+    borderRadius('!rounded-lg'),
```
Comment (T-Damer) [inline src/components/ToolTip.tsx:46]: Do we need important here if we don't override original library?

File: src/components/ToolTip.tsx [hunk 4]
```diff
@@ -1,45 +1,91 @@
...
+const tooltipClasses = (mobile: boolean, show: boolean) => {
+  return classnames(
+    position('absolute'),
     backgroundColor('bg-formal-accent'),
-    maxWidth('max-w-sm'),
-    mobile ? inset('!left-0', '!right-0') : undefined,
-    opacity('!opacity-100'),
-    borderRadius('!rounded-lg')
+    inset(mobile ? 'left-0' : 'left-3', mobile ? 'right-0' : 'right-3'),
+    inset('bottom-4'),
```
Comment (T-Damer) [inline src/components/ToolTip.tsx:42]: Let's move this inset(content) inside another

File: src/components/ToolTip.tsx [hunk 5]
```diff
@@ -1,45 +1,91 @@
...
+          className={tooltipClasses(xs, isShow)}
+          style={{ visibility: isShow ? 'visible' : 'collapse' }}
+        >
+          <HighlightedText>{text}</HighlightedText>
+          <div className={triangle}></div>
+        </div>
+      </div>
+      <div
+        ref={childrenRef}
+        className={width('w-full')}
```
Comment (T-Damer) [inline src/components/ToolTip.tsx:82]: Let's extract this into `variable` so we can change it fast if needed

File: src/hooks/useClickOutside.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { MutableRef } from 'preact/hooks'
+import { useEffect } from 'react'
+
+export default function (
+  ref: MutableRef<HTMLDivElement>,
+  callback: () => void
+) {
+  useEffect(() => {
+    function handleClickOutside(event: Event) {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
```
Comment (T-Damer) [inline src/hooks/useClickOutside.ts:10]: You can go without `{}` here

File: src/hooks/useClickOutside.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        callback()
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => {
```
Comment (T-Damer) [inline src/hooks/useClickOutside.ts:16]: And here

**Review by T-Damer — CHANGES_REQUESTED** [review]
Need a little code cleaning, otherwise looks pretty good

**Review by T-Damer — APPROVED** [review]
lgtm, great work :)

**Pull request merged**